packages/wallet/test/hardware/ledger/LedgerKeyAgent.test.ts

```diff
@@ -10,298 +10,382 @@ import {
 import { AssetId, createStubStakePoolProvider, mockProviders as mocks } from '@cardano-sdk/util-dev';
 import { BaseWallet, createPersonalWallet } from '../../../src';
 import { Cardano, Serialization } from '@cardano-sdk/core';
+import { HID } from 'node-hid';
 import { Hash32ByteBase16 } from '@cardano-sdk/crypto';
 import { HexBlob } from '@cardano-sdk/util';
 import { InitializeTxProps, InitializeTxResult } from '@cardano-sdk/tx-construction';
 import { LedgerKeyAgent, LedgerTransportType } from '@cardano-sdk/hardware-ledger';
 import { firstValueFrom } from 'rxjs';
+import { getDevices } from '@ledgerhq/hw-transport-node-hid-noevents';
 import { dummyLogger as logger } from 'ts-log';
 import { mockKeyAgentDependencies } from '../../../../key-management/test/mocks';
 import DeviceConnection from '@cardano-foundation/ledgerjs-hw-app-cardano';
 
+const getHidDevice = () => {
+  const ledgerDevicePath = getDevices()[0]?.path;
+  if (!ledgerDevicePath) {
+    throw new Error('No ledger device connected');
+  }
+  return new HID(ledgerDevicePath);
+};
+
+const cleanupEstablishedConnections = async () => {
+  if (LedgerKeyAgent.deviceConnections.length === 0) return;
+  for (const { deviceConnection } of LedgerKeyAgent.deviceConnections) {
+    await deviceConnection.transport.close();
+  }
+  LedgerKeyAgent.deviceConnections = [];
+};
+
 describe('LedgerKeyAgent', () => {
-  let ledgerKeyAgent: LedgerKeyAgent;
```

Comment:
I reorganized the structure slightly. I moved all the code and tests related to creating `LedgerKeyAgent` instance and using it together with `PersonalWallet` under the dedicated describe so it does not hang around while executing other tests.